packages/ra-data-simple-rest/src/index.ts

```diff
@@ -10,7 +10,78 @@ import {
     UPDATE_MANY,
     DELETE,
     DELETE_MANY,
-} from 'react-admin';
+} from 'ra-core';
+
+interface Pagenation {
+    page: number;
+    perPage: number;
+}
+
+interface Sort {
+    field: string;
+    order: 'ASC' | 'DESC';
+}
+
+interface GetListRequestParams {
+    pagination: Pagenation;
+    sort: Sort;
+    filter: {
+        [key: string]: any;
+    };
+}
+
+interface IdParams {
+    id: number | string;
+}
+interface IdsParams {
+    ids: (number | string)[];
+}
+
+interface GetOneRequestParams extends IdParams {}
```

Comment:
these types should be added to ra-core as they'll be required for all data providers.

Reply:
I agree.I'll move this.